File: src/dataset/javascript/robot-name_exercism-ThomasZumsteg/gemini/GeminiFlash_robot-name_v1.js
var Robot = function() { this.name = getName() }

Robot.prototype.reset = function() {
	this.name = getName();
};

// Keeps track of robot names in use
var namesInUse = new Set();
const MAX_ROBOTS = 676000; // Maximum possible robot names

function getName() {
	/* Generates a robot name */
	if (namesInUse.size >= MAX_ROBOTS) {
        throw new Error("All possible robot names have been used.");
    }

	let name;
	do {
		name = generateName();
	} while (namesInUse.has(name));

	namesInUse.add(name);
	return name;
}

function generateName() {
    let result = '';
    for (let i = 0; i < 2; i++) {
        result += String.fromCharCode(Math.floor(Math.random() * 26) + 65); // A-Z
    }
    for (let i = 0; i < 3; i++) {
        result += String.fromCharCode(Math.floor(Math.random() * 10) + 48); // 0-9
    }
    return result;
}

module.exports = Robot;

File: src/dataset/python/robot-name_exercism-fruit-in/gemini/GeminiFlash_robot_name_v1.py
import random
import string


class Robot:
    _existing_names = set()

    def __init__(self):
        self.name = self._generate_name()

    def _generate_name(self):
        while True:
            name = ''.join(random.choices(string.ascii_uppercase, k=2) +
                           random.choices(string.digits, k=3))
            if name not in Robot._existing_names:
                Robot._existing_names.add(name)
                return name

    def reset(self):
        self.name = self._generate_name()